### PR TITLE
Merge pull request #33 from LastOfUs13/master

### PR DESCRIPTION
**Pull request by LastOfUs13**
Merge pull request #32 from LastOfUs13/HW14

### CODE REVIEW

File: HW14/proxy/txt_proxy.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        if self.__is_actual:
+            return self.__result
+        else:
+            self.__result = self.__reader.read_file()
+            self.__is_actual = True
+            return self.__result
+
+    def write_file(self, user_data):
+        self.__result = self.__writer.write_file(user_data)
+        self.__is_actual = True
```
Comment (bohdandovbsyh) [inline HW14/proxy/txt_proxy.py]: where is proxy? 
you write to file every time 

and why did you change actual to True after writing?

**Review by bohdandovbsyh — APPROVED** [review]

File: HW14/proxy/txt_proxy.py (hunk 1 above)
Comment (LastOfUs13) [reply HW14/proxy/txt_proxy.py]: sorry,at the beginning ,i didn't catch the main idea- is it better?                                                                                     
      def write_file(self, user_data):
          if not self.__is_actual:
              self.__result = user_data
              self.__is_actual = True
         else:
              self.__writer.write_file(user_data)
              return self.__result

Comment (bohdandovbsyh) [reply HW14/proxy/txt_proxy.py]: No. else is okay. 
But think about logic in the first if statement

File: HW14/proxy/txt_proxy.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        if self.__is_actual:
+            return self.__result
+        else:
+            self.__result = self.__reader.read_file()
+            self.__is_actual = True
+            return self.__result
+
+    def write_file(self, user_data):
+        if user_data == self.__result:
+            self.__is_actual = False
```
Comment (bohdandovbsyh) [inline HW14/proxy/txt_proxy.py:23]: No! in this case just do Nthng. Why do you change actual if you do not change data

File: HW14/proxy/txt_proxy.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        else:
+            self.__writer.write_file(user_data)
```
Comment (bohdandovbsyh) [inline HW14/proxy/txt_proxy.py:25]: after writing you should change __is_actual because data was changed
